src/index.js

```diff
@@ -419,14 +419,20 @@ export default React.createClass({
               <span key={col.id}
                 style={{display: 'flex', alignContent: 'flex-end', flex: 1}}>
                 {!col.hideFilter ? (
-                  <input type='text'
-                    style={{
-                      flex: 1,
-                      width: 20
-                    }}
-                    value={filter ? filter.value : ''}
-                    onChange={(event) => this.filterColumn(column, event, col)}
-                  />
+                  _.normalizeComponent(col.filterRender,
+                    {
+                      onFilterChange: (value) => (this.filterColumn(column, value, col))
+                    },
+                    (
+                      <input type='text'
```

Comment:
So, instead of defining the default component inline in index.js, you could set the default filter component for every column here: https://github.com/tannerlinsley/react-table/blob/master/src/defaultProps.js#L107, like this:

```javascript
column: {
  ...
  filterRender: ({
    filter,
    onFilterChange
  }) => (
    <input type='text'
      style={{
        flex: 1
      }}
      value={filter ? filter.value : ''}
      onChange={(e) => onFilterChange(event.target.value)}
    />
  )
}
```

Then in index.js, you can use it like this:
```javascript
_.normalizeComponent(col.filterRender, {
  onFilterChange: value => this.filterColumn(column, value, col)
}
//...
_.normalizeComponent(col.filterRender, {
  onFilterChange: value => this.filterColumn(column, value)
```